exchanges/subscription/subscription.go

```diff
@@ -0,0 +1,90 @@
+package subscription
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/kline"
+)
+
+// DefaultKey is the fallback key for AddSuccessfulSubscriptions
+type DefaultKey struct {
+	Channel string
+	Pair    currency.Pair
+	Asset   asset.Item
+}
+
+// State tracks the status of a subscription channel
+type State uint8
+
+const (
+	UnknownState       State = iota // UnknownState subscription state is not registered, but doesn't imply Inactive
+	SubscribingState                // SubscribingState means channel is in the process of subscribing
+	SubscribedState                 // SubscribedState means the channel has finished a successful and acknowledged subscription
+	UnsubscribingState              // UnsubscribingState means the channel has started to unsubscribe, but not yet confirmed
+
+	TickerChannel    = "ticker"    // TickerChannel Subscription Type
+	OrderbookChannel = "orderbook" // OrderbookChannel Subscription Type
+	CandlesChannel   = "candles"   // CandlesChannel Subscription Type
+	AllOrdersChannel = "allOrders" // AllOrdersChannel Subscription Type
+	AllTradesChannel = "allTrades" // AllTradesChannel Subscription Type
+	MyTradesChannel  = "myTrades"  // MyTradesChannel Subscription Type
+	MyOrdersChannel  = "myOrders"  // MyOrdersChannel Subscription Type
+)
+
+// Subscription container for streaming subscriptions
+type Subscription struct {
+	Key           any                    `json:"-"`
```

Comment:
I think the main thing I'd want to see is `Enabled` as a field. The way that this is placed into the config makes it difficult to re-add/re-enable a subscription because it requires knowledge of the exchange's websocket implementation code eg I need to know that kucoin's ticker sub is formatted as `/market/ticker:%s` otherwise it won't work, which isn't as friendly as abstracting it. 

By having an enabled field, you can disable all subscriptions - which would allow an external application to completely manage their own subscriptions too

Just a quick note I'm not requesting you create such abstractions for the config, but I think an enabled field is a good middleground to minimise the scope of this initial PR.

Reply:
Agreed, I like that. Thank you.
On it.

Reply:
Fixed 8a2e600